Kernel/Net/Routing.h

```diff
@@ -2,4 +2,12 @@
 
 #include <Kernel/Net/NetworkAdapter.h>
 
-WeakPtr<NetworkAdapter> adapter_for_route_to(const IPv4Address&);
+struct RoutingDecision
+{
+    WeakPtr<NetworkAdapter> adapter;
+    const MACAddress& next_hop;
```

Comment:
This seems very wrong. This makes it possible to return a `RoutingDecision` whose `next_hop` refers to a now-invalid stack variable.

Reply:
Good point - another victim of my beginner-level understanding of C++ references!